source/includes/fundamentals/code-snippets/IndexPage.java

```diff
@@ -0,0 +1,150 @@
+package docs.indexes;
+
+import com.mongodb.MongoCommandException;
+import com.mongodb.client.*;
+import com.mongodb.client.model.Indexes;
+import com.mongodb.client.model.Sorts;
+import com.mongodb.client.model.geojson.Point;
+import com.mongodb.client.model.geojson.Position;
+import org.apache.log4j.BasicConfigurator;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+
+import static com.mongodb.client.model.Filters.*;
+import static com.mongodb.client.model.Projections.*;
+
+public class IndexPage {
+
+    private final MongoClient mongoClient;
+    private final MongoDatabase database;
+    private MongoCollection<Document> collection;
+
+    private IndexPage() {
+        BasicConfigurator.configure();
+
+        final String uri = "mongodb+srv://<atlas-uri>/<dbname>?retryWrites=true&w=majority";
+
+        mongoClient = MongoClients.create(uri);
+        database = mongoClient.getDatabase("sample_mflix");
+        collection = database.getCollection("movies");
+    }
+
+    public static void main(String[] args) {
+        IndexPage page = new IndexPage();
+        page.singleIndex();
+        page.compoundIndex();
+        page.multiKeyIndex();
+        page.textIndex();
+        page.geoSpatialIndex();
+    }
+
+    private void singleIndex() {
+        System.out.println("single index");
+        // begin single index
+        String resultCreateIndex = collection.createIndex(Indexes.ascending("title"));
+        System.out.println(String.format("Index created: %s", resultCreateIndex));
```

Comment:
Is there a style reason you'd use String.format here instead of printf, or just personal choice? I'd think that printf would be easier to understand for intermediate-level developers.

Reply:
I used this to avoid having to do `System.out.printf("Index created: %s%n", resultCreateIndex);` for the newline.